jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java

```diff
@@ -582,18 +582,59 @@ public boolean onRemoveField(HttpField field)
                         {
                             _contentType = null;
                             _mimeType = null;
-                            _characterEncoding = switch (_encodingFrom)
+                            if (!isWriting())
                             {
-                                case SET_CHARACTER_ENCODING, SET_LOCALE -> _characterEncoding;
-                                default -> null;
-                            };
+                                _characterEncoding = switch (_encodingFrom)
+                                {
+                                    case SET_CHARACTER_ENCODING, SET_LOCALE -> _characterEncoding;
+                                    default -> null;
+                                };
+                            }
                         }
                     }
                 }
             }
             return true;
         }
 
+        /**
+         *
+         */
+        @Override
+        public HttpField onReplaceField(HttpField oldField, HttpField newField)
+        {
+            if (oldField == null)
+            {
+                return onAddField(newField);
+            }
+
+            if (newField == null)
+            {
+                onRemoveField(oldField);
+                return null;
+            }
+
+            switch (newField.getHeader())
+            {
+                case CONTENT_LENGTH ->
+                {
+                    if (!isCommitted())
+                    {
+                        return setContentLength(newField);
+                    }
+                }
+                case CONTENT_TYPE ->
+                {
+                    if (!isCommitted())
+                    {
+                        return setContentType(newField);
+                    }
```

Comment:
ditto
```suggestion
                    return isCommitted() ? null : setContentType(newField);
```

Oh no, and all the add and remove ones are wrong as well!